src/main/fc/cli.c

```diff
@@ -3009,48 +3012,259 @@ static void cliResource(char *cmdline)
+            cliPrintLinef(" %s", owner);
+        }
+    }
+}
+
+static void cliDma(char* cmdLine)
+{
+    UNUSED(cmdLine);
+    printDma();
+}
 #endif /* USE_RESOURCE_MGMT */
 
+#ifdef USE_TIMER_MGMT
+
+static void printTimer(uint8_t dumpMask)
+{
+    char buffer[12];
+    
+    cliPrintLine("# examples: ");
+    const char *format = "timer TIM%d_CH%d%s %c%02d AF%d %s";
+    cliPrint("#");
+#if defined(STM32F7) || defined(STM32F4)
+    const char *dmaFormat = "DMA%d_ST%d CH%d";
+    cliPrintLinef(format, 1, 1, "", 'A', 1, 1, "DMA1_ST1 CH1 (or NONE)");
+#elif defined(STM32F3)
+    const char *dmaFormat = "DMA%d_CH%d";
+    cliPrintLinef(format, 1, 1, "", 'A', 1, 1, "DMA1_CH1 (or NONE)");
+#else
+#error "No format specified for this CPU for outputting timer values"
+#endif
+    const char *formatUnassigned = "timer TIM%d_CH%d NONE";
+    cliPrint("#");
+    cliPrintLinef(formatUnassigned, 1, 1);
+    
+    for (unsigned int i = 0; i < TIMER_CHANNEL_COUNT; i++) {
+
+        const timerTag_t timerTag = timerTags[i];
+        const uint8_t timer = TIMER_TAG_INDEX(timerTag);
+        const uint8_t channel = TIMER_TAG_CHANNEL_INDEX(timerTag);
+        const bool inverted = timerChannelConfig(i)->inverted;
+        const ioTag_t ioTag = timerChannelConfig(i)->ioTag;
+
+        if (!ioTag) {
+            if (!(dumpMask & HIDE_UNUSED)) {
+                cliDumpPrintLinef(dumpMask, false, formatUnassigned, timer, channel);
+            }
+        } else {
+            memset(buffer, 0, sizeof(buffer));
+            if (timerChannelConfig(i)->dma) {
+                tfp_sprintf(buffer,
+                    dmaFormat, 
+                    DMA_DEVICE(timerChannelConfig(i)->dma),
+                    DMA_DEVICE_INDEX(timerChannelConfig(i)->dma)
+#if defined(STM32F7) || defined(STM32F4)
+                    , timerChannelConfig(i)->dmaChannel
+#endif
+                    );
+            } else {
+                tfp_sprintf(buffer, "%s", "NONE");
+            }
+            
+            cliDumpPrintLinef(dumpMask, false, format, 
+                timer, 
+                channel,
+                (inverted) ? "N" : "",
+                IO_GPIOPortIdxByTag(ioTag) + 'A', IO_GPIOPinIdxByTag(ioTag),
+                timerChannelConfig(i)->alternateFunction,
+                buffer
+                );
+        }
+    }
+}
+
+static void cliTimer(char *cmdline)
+{
+    int len = strlen(cmdline);
+
+    if (len == 0) {
+        printTimer(DUMP_MASTER | HIDE_UNUSED);
+        return;
+    } else if (strncasecmp(cmdline, "list", len) == 0) {
+        printTimer(DUMP_MASTER);
+        return;
+    }
+    
+    char *pch = NULL;
+    char *saveptr;
+    char buffer[20];
+    
+    memset(buffer, 0, sizeof(buffer));
+    timerTag_t timerTag = 0;
+    
+    const char *timChformat = "tim%d_ch%d%s";
+    bool inverted = false;
+    int timerChannelIndex = 0;
+    
+    pch = strtok_r(cmdline, " ", &saveptr);
+    for (int i = 0; i < TIMER_CHANNEL_COUNT; i++) {
```

Comment:
Sorry that's not a very constructive comment at all. What's ugly? Why is it ugly? Do you have a suggested better alternative? Or a better approach?

Merely stating it is an ugly hack provides absolutely no value to anyone. No one learns from it. I'm keen to hear your valuable constructive thoughts, as I'm always interested in learning.